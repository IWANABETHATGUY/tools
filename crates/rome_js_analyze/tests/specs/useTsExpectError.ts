// invalid
// @ts-ignore
let foo: boolean = 1;
// @ts-ignore: Blah blah blah
let foo: boolean = 1;
/* @ts-ignore */
let foo: boolean = 1;
/** @ts-ignore */
let foo: boolean = 1;
/**
 * @ts-ignore */
let foo: boolean = 1;
/**
 ** @ts-ignore */
let foo: boolean = 1;
/**
 ** @ts-ignore
 ** @ts-ignore */
let foo2: boolean = 1;

// valid
// @ts-expect-error
let foo: boolean = 1;
// @ts-expect-error: Blah blah blah
let foo: boolean = 1;
/* @ts-expect-error */
let foo: boolean = 1;
/** @ts-expect-error */
let foo: boolean = 1;
/**
 * @ts-expect-error */
let foo: boolean = 1;
/**
 ** @ts-expect-error */
let foo: boolean = 1;
