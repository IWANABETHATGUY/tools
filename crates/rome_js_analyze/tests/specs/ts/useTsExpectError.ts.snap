---
source: crates/rome_js_analyze/tests/spec_tests.rs
assertion_line: 88
expression: useTsExpectError.ts
---
# Input
```js
// invalid
// @ts-ignore
let foo: boolean = 1;
// @ts-ignore: Blah blah blah
let foo: boolean = 1;
/* @ts-ignore */
let foo: boolean = 1;
/** @ts-ignore */
let foo: boolean = 1;
/**
 * @ts-ignore */
let foo: boolean = 1;
/**
 ** @ts-ignore */
let foo: boolean = 1;
/**
 ** @ts-ignore
 ** @ts-ignore */
let foo2: boolean = 1;

// valid
// @ts-expect-error
let foo: boolean = 1;
// @ts-expect-error: Blah blah blah
let foo: boolean = 1;
/* @ts-expect-error */
let foo: boolean = 1;
/** @ts-expect-error */
let foo: boolean = 1;
/**
 * @ts-expect-error */
let foo: boolean = 1;
/**
 ** @ts-expect-error */
let foo: boolean = 1;

```

# Diagnostics
```
warning[ts/useTsExpectError]: Prefer @ts-expect-error to get notified when suppression is no longer necessary.
  ┌─ useTsExpectError.ts:3:1
  │
3 │ let foo: boolean = 1;
  │ ---------------------

Suggested fix: Use `@ts-expect-error` instead of `@ts-ignore`.
    | @@ -1,5 +1,5 @@
0 0 |   // invalid
1   | - // @ts-ignore
  1 | + // @ts-expect-error
2 2 |   let foo: boolean = 1;
3 3 |   // @ts-ignore: Blah blah blah
4 4 |   let foo: boolean = 1;


```

```
warning[ts/useTsExpectError]: Prefer @ts-expect-error to get notified when suppression is no longer necessary.
  ┌─ useTsExpectError.ts:5:1
  │
5 │ let foo: boolean = 1;
  │ ---------------------

Suggested fix: Use `@ts-expect-error` instead of `@ts-ignore`.
    | @@ -1,7 +1,7 @@
0 0 |   // invalid
1 1 |   // @ts-ignore
2 2 |   let foo: boolean = 1;
3   | - // @ts-ignore: Blah blah blah
  3 | + // @ts-expect-error: Blah blah blah
4 4 |   let foo: boolean = 1;
5 5 |   /* @ts-ignore */
6 6 |   let foo: boolean = 1;


```

```
warning[ts/useTsExpectError]: Prefer @ts-expect-error to get notified when suppression is no longer necessary.
  ┌─ useTsExpectError.ts:7:1
  │
7 │ let foo: boolean = 1;
  │ ---------------------

Suggested fix: Use `@ts-expect-error` instead of `@ts-ignore`.
    | @@ -3,7 +3,7 @@
2 2 |   let foo: boolean = 1;
3 3 |   // @ts-ignore: Blah blah blah
4 4 |   let foo: boolean = 1;
5   | - /* @ts-ignore */
  5 | + /* @ts-expect-error */
6 6 |   let foo: boolean = 1;
7 7 |   /** @ts-ignore */
8 8 |   let foo: boolean = 1;


```

```
warning[ts/useTsExpectError]: Prefer @ts-expect-error to get notified when suppression is no longer necessary.
  ┌─ useTsExpectError.ts:9:1
  │
9 │ let foo: boolean = 1;
  │ ---------------------

Suggested fix: Use `@ts-expect-error` instead of `@ts-ignore`.
    | @@ -5,7 +5,7 @@
4 4 |   let foo: boolean = 1;
5 5 |   /* @ts-ignore */
6 6 |   let foo: boolean = 1;
7   | - /** @ts-ignore */
  7 | + /** @ts-expect-error */
8 8 |   let foo: boolean = 1;
9 9 |   /**
10 10 |    * @ts-ignore */


```

```
warning[ts/useTsExpectError]: Prefer @ts-expect-error to get notified when suppression is no longer necessary.
   ┌─ useTsExpectError.ts:12:1
   │
12 │ let foo: boolean = 1;
   │ ---------------------

Suggested fix: Use `@ts-expect-error` instead of `@ts-ignore`.
      | @@ -8,7 +8,7 @@
 7  7 |   /** @ts-ignore */
 8  8 |   let foo: boolean = 1;
 9  9 |   /**
10    | -  * @ts-ignore */
   10 | +  * @ts-expect-error */
11 11 |   let foo: boolean = 1;
12 12 |   /**
13 13 |    ** @ts-ignore */


```

```
warning[ts/useTsExpectError]: Prefer @ts-expect-error to get notified when suppression is no longer necessary.
   ┌─ useTsExpectError.ts:15:1
   │
15 │ let foo: boolean = 1;
   │ ---------------------

Suggested fix: Use `@ts-expect-error` instead of `@ts-ignore`.
      | @@ -11,7 +11,7 @@
10 10 |    * @ts-ignore */
11 11 |   let foo: boolean = 1;
12 12 |   /**
13    | -  ** @ts-ignore */
   13 | +  ** @ts-expect-error */
14 14 |   let foo: boolean = 1;
15 15 |   /**
16 16 |    ** @ts-ignore


```

```
warning[ts/useTsExpectError]: Prefer @ts-expect-error to get notified when suppression is no longer necessary.
   ┌─ useTsExpectError.ts:19:1
   │
19 │ let foo2: boolean = 1;
   │ ----------------------

Suggested fix: Use `@ts-expect-error` instead of `@ts-ignore`.
      | @@ -15,7 +15,7 @@
14 14 |   let foo: boolean = 1;
15 15 |   /**
16 16 |    ** @ts-ignore
17    | -  ** @ts-ignore */
   17 | +  ** @ts-expect-error */
18 18 |   let foo2: boolean = 1;
19 19 |   
20 20 |   // valid


```


